test/Tizen.NUI.Scene3D.Sample/Scene3DSample.cs

```diff
@@ -0,0 +1,245 @@
+        mModelRotateAnimation.AnimateBy(mModel, "Orientation", new Rotation(new Radian(new Degree(360.0f)), Vector3.YAxis));
+
+        mModelRotateAnimation.Looping = true;
+        mModelRotateAnimation.Play();
+
+        mSceneView.Add(mModel);
+    }
+
+    void SetupIBLimage(string specularUrl, string diffuseUrl, float iblFactor)
+    {
+        mSceneView.SetImageBasedLightSource(IMAGE_DIR + specularUrl, IMAGE_DIR + diffuseUrl,iblFactor);
+    }
+
+    void OnKeyEvent(object source, Window.KeyEventArgs e)
+    {
+        if (e.Key.State == Key.StateType.Down)
+        {
+            FullGC();
+            //Streamline.AnnotateChannelEnd(0);
+
+            switch( e.Key.KeyPressedName )
+            {
+                case "Escape":
+                case "Back":
+                {
+                    Deactivate();
+                    Exit();
+                    break;
+                }
+
+                case "Return":
+                case "Select":
+                {
+                    currentCameraIndex++;
+                    if(currentCameraIndex >= mSceneView.GetCameraCount())
+                    {
+                        currentCameraIndex = 0;
+                    }
+
+                    mSceneView.CameraTransition(currentCameraIndex, cameraAnimationDurationMilliSeconds);
+                    break;
+                }
+
+                case "1":
+                {
+                    currentModelIndex++;
+                    if(currentModelIndex >= ModelUrlList.Count)
+                    {
+                        currentModelIndex = 0;
+                    }
+
+                    CreateModel(ModelUrlList[currentModelIndex]);
+                    break;
+                }
+                case "2":
+                {
+                    currentIBLIndex++;
+                    if(currentIBLIndex >= IBLUrlList.Count)
+                    {
+                        currentIBLIndex = 0;
+                    }
+
+                    SetupIBLimage(IBLUrlList[currentIBLIndex].Item1, IBLUrlList[currentIBLIndex].Item2, 0.7f);
+                    break;
+                }
+            }
+        }
+    }
+
+    private static readonly List<string> ModelUrlList = new List<string>()
+    {
+        // Model reference : https://sketchfab.com/models/b81008d513954189a063ff901f7abfe4
+        // Get from KhronosGroup https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/DamagedHelmet
+        "DamagedHelmet/DamagedHelmet.gltf",
+        "BoxAnimated/BoxAnimated.gltf",
+    };
+    private int currentModelIndex = 0;
+
+    private static readonly List<(string, string)> IBLUrlList = new List<(string, string)>
+    {
+        ("forest_diffuse_cubemap.png", "forest_specular_cubemap.png"),
+        ("papermill_E_diffuse-64.ktx", "papermill_pmrem.ktx"),
+        ("Irradiance.ktx", "Radiance.ktx"),
+    };
+    private int currentIBLIndex = 0;
```

Comment:
How about these variables on the top of the class?
Are there any reasons to define these indices and lists in the middle of the class?

Reply:
I move them top of this demo, and make region.